### ISSUE TITLE
Receiving transfers fail with EventUnlockClaimFailed

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/raiden#2213]
## Problem Definition

Something bad happened with receiving transfers from the echo node, which triggered the following error to be logged on every block:
```
raiden_1  | 2018-08-20 09:50:32 [debug    ] new block                      [raiden.tasks] number=3873824
raiden_1  | 2018-08-20 09:50:32 [debug    ] STATE CHANGE                   [raiden.raiden_service] node=008135fa state_change=<Block 3873824>
raiden_1  | 2018-08-20 09:50:32 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<ContractSendSecretReveal secrethash:b'8\xda=\xee\xe2Ej\xe4\xa3\xc07\xeb 8f\x82\xda\x91\x88L7\x0b\x0b\xc9\xe3M\x8a\x0f\xd9\x9bf:'>
raiden_1  | 2018-08-20 09:50:32 [info     ] secret 0x38da3deee2456ae4a3c037eb20386682da91884c370b0bc9e34d8a0fd99b663a already registered. [raiden.network.proxies.secret_registry] contract=3de6b821 node=008135fa secrethash=0x38da3deee2456ae4a3c037eb20386682da91884c370b0bc9e34d8a0fd99b663a
raiden_1  | 2018-08-20 09:50:32 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<EventUnlockClaimFailed id:1342 secrethash:b2a4ddf8 reason:lock expired>
raiden_1  | 2018-08-20 09:50:32 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<EventUnlockClaimFailed id:1534519337516 secrethash:8e49d80b reason:lock expired>
raiden_1  | 2018-08-20 09:50:32 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<EventUnlockClaimFailed id:1534519432806 secrethash:bd017ae4 reason:lock expired>
raiden_1  | 2018-08-20 09:50:32 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<EventUnlockClaimFailed id:1534519895390 secrethash:58c661e4 reason:lock expired>
```

Every time I receive a transfer from this node, the transfer fails and I get one of those, and after it expires, it joins this list of errors which is thrown on every block:

```
...
raiden_1  | 2018-08-20 09:50:37 [info     ] MESSAGE RECEIVED               [raiden.network.transport.matrix] current_user=@0x008135fa6977849add47ba058929ca2096725466:transport01.raiden.network message=<LockedTransfer [chainid:3 msgid:3257454750105885723 paymentid:1534758631142 token_network:9fe1f19d channel_identifier:64 nonce:18 transferred_amount:17 locked_amount:11 locksroot:5f0e65b8 hash:66cdb267 secrethash:4a95edf1 expiration:3874325 amount:2]> node=008135fa sender=02f4b6bc
raiden_1  | 2018-08-20 09:50:37 [debug    ] HEALTHCHECK                    [raiden.network.transport.matrix] current_user=@0x008135fa6977849add47ba058929ca2096725466:transport01.raiden.network peer_address=0x02f4b6bc65561a792836212ebc54434db0ab759a
raiden_1  | 2018-08-20 09:50:37 [debug    ] STATE CHANGE                   [raiden.raiden_service] node=008135fa state_change=<ActionInitTarget route:<RouteState hop:02f4b6bc channel_identifier:64> transfer:<LockedTransferSignedState msgid:3257454750105885723 id:1534758631142 token:0x0f114a1e9db192502e7856309cc899952b3db1ed lock:<HashTimeLockState amount:2 expiration:3874325 secrethash:4a95edf1> target:0x008135fa6977849add47ba058929ca2096725466>>
raiden_1  | 2018-08-20 09:50:37 [debug    ] RAIDEN EVENT                   [raiden.raiden_service] node=008135fa raiden_event=<EventUnlockClaimFailed id:1534758631142 secrethash:4a95edf1 reason:Invalid LockedTransfer message. Nonce did not change sequentially, expected: 13 got: 18.>
```

[raiden-debug.log](https://github.com/raiden-network/raiden/files/2301966/raiden-debug.log) <= the whole session log

### System Description

Current master: 68fc3dcedd830014d31dd083b311dd3426d49f17

**Comment by andrevmatos** [comment]
In some initial tests @hackaugusto and I did, it seems like the messages were sent or handled out of order (nonce=12, unlock message, arrived after nonce=13, transfer), so nonce=13 failed, and somehow was not retried afterwards, while the nonce was still updated in the sender, which kept increasing it on every message (as expected, but it'd depend on the retries to eventually succeed).
On #2208, as part of exception handling of children/retry greenlets, some work was done to simplify matrix retries, but there may still be work to do.

**Comment by jomuel** [comment]
Thanks. I still cannot reproduce the bug but this should be enough to create a failing test instead, if your changes in #2208 didn't fix it.

**Comment by LefterisJP** [comment]
So the task now @jomuel is to create a test for this issue? And then we can fix and close?

**Comment by jomuel** [comment]
Yes that's what I'd suggest...

**Comment by jomuel** [comment]
The test has been included into pr #2268. With the changes in said pr the test passes, so we can close this issue now.
